client/my-sites/email/inbox/index.js

```diff
@@ -33,26 +36,48 @@ const InboxManagement = ( { domains } ) => {
 		);
 	}
 
-	const domainsWithSubscriptions = domains.filter(
-		( domain ) => ! domain.isWPCOMDomain && hasPaidEmailWithUs( domain )
+	// Find out if the site has at least one mailbox in the site
+	const hasAnyMailboxes = domains.some(
+		( domain ) =>
+			( ! domain.isWPCOMDomain &&
+				// it may be worth also checking for has(EmailProduct)WithUs( domain ) for each branch below
+				getGSuiteMailboxCount( domain ) > 0 ) ||
+			getConfiguredTitanMailboxCount( domain ) > 0
 	);
 
-	//EmailManagementHome logic will handle the case where there is only one domain to show directly the email comparison
-	//and also if there is no domain so, it will show a CTA to buy domain
-	if ( domainsWithSubscriptions.length === 0 ) {
+	// Load the Inbox UI if at least one mailbox occurs
+	if ( hasAnyMailboxes ) {
+		return <MailboxSelectionList />;
+	}
+
+	const domainsWithEmailServices = domains.filter(
+		( domain ) =>
+			! domain.isWPCOMDomain && ( hasPaidEmailWithUs( domain ) || hasEmailForwards( domain ) )
+	);
+
+	const showActiveDomainList = domainsWithEmailServices.length > 0;
+
+	if ( showActiveDomainList ) {
 		return (
 			<CalypsoShoppingCartProvider>
 				<EmailManagementHome
 					emailListInactiveHeader={ getMainHeader() }
 					sectionHeaderLabel={ translate( 'Domains' ) }
 					showActiveDomainList={ false }
-				/>
+				></EmailManagementHome>
 			</CalypsoShoppingCartProvider>
 		);
 	}
 
-	//If we are at this point it means that we've at least have one subscription to show in mailbox selector
-	return <MailboxSelectionList />;
+	// Delegate to <EmailManagementHome/> in case we need to upsell a domain/mailbox
+	return (
+		<CalypsoShoppingCartProvider>
+			<EmailManagementHome
+				emailListInactiveHeader={ getMainHeader() }
+				sectionHeaderLabel={ translate( 'Domains' ) }
+			/>
+		</CalypsoShoppingCartProvider>
+	);
```

Comment:
Rather than this if construct, it may be better to bind the boolean `showActiveDomainList` to the apt, and similarly named prop.

```js
return (
		<CalypsoShoppingCartProvider>
			<EmailManagementHome
				emailListInactiveHeader={ getMainHeader() }
				sectionHeaderLabel={ translate( 'Domains' ) }
                showActiveDomainList={ showActiveDomainList }
			/>
		</CalypsoShoppingCartProvider>
	);
```